pkg/platform/apiserver/versionproxy.go

```diff
@@ -0,0 +1,51 @@
+package apiserver
+
+import (
+	"net/http"
+
+	"github.com/emicklei/go-restful"
+	"k8s.io/client-go/rest"
+	platforminternalclient "tkestack.io/tke/api/client/clientset/internalversion/typed/platform/internalversion"
+	"tkestack.io/tke/pkg/platform/proxy"
+)
+
+type VersionProxyHandler struct {
+	config *rest.Config
+}
+
+func NewVersionProxyHandler(config *rest.Config) *VersionProxyHandler {
+	return &VersionProxyHandler{
+		config: config,
+	}
+}
+
+func (s *VersionProxyHandler) Install(c *restful.Container) {
+	versionWS := new(restful.WebService)
+	versionWS.Path("/version")
+	versionWS.Doc("git code version from which this is built")
+	versionWS.Route(
+		versionWS.GET("/").To(s.handle).
+			Doc("get the code version").
+			Operation("getVersion").
+			Produces(restful.MIME_JSON).
+			Consumes(restful.MIME_JSON))
+
+	c.Add(versionWS)
+}
+
+func (s *VersionProxyHandler) handle(req *restful.Request, resp *restful.Response) {
+	platformClient := platforminternalclient.NewForConfigOrDie(s.config)
+	client, err := proxy.ClientSet(req.Request.Context(), platformClient)
+	if err != nil {
+		resp.WriteHeaderAndEntity(http.StatusBadRequest, "err1"+err.Error())
+		return
```

Comment:
why `err1` rather than "err.Error()"

Reply:
to keep compatibility, please add logic to check if token or client crt/key pass to back-end, then proxy the request to remote k8s server, otherwise get version info from local